src/Data/Array/Accelerate/Data/Monoid.hs

```diff
@@ -198,3 +198,31 @@ instance (Elt a, Elt b, Elt c, Elt d, Elt e, Monoid (Exp a), Monoid (Exp b), Mon
                 in
                 lift (a1 `mappend` a2, b1 `mappend` b2, c1 `mappend` c2, d1 `mappend` d2, e1 `mappend` e2)
 
+#if __GLASGOW_HASKELL__ >= 800
```

Comment:
I followed the existing convention in this module of conditionally defining `Semigroup` instances using CPP. If you'd prefer, we could axe this CPP by depending on the `semigroups` library for backwards compatibility with versions of GHC older than 8.0.

Reply:
Oops, I just discovered there's a `Data.Array.Accelerate.Data.Semigroup` module which already implements this. In that case, the fix is much simpler: just import that here.